src/app/lib/utilities/preprocessor/rules/addIdsToItems.js

```diff
@@ -0,0 +1,31 @@
+import uuid from 'uuid';
+import compose from 'ramda/src/compose';
+import pathOr from 'ramda/src/pathOr';
+
+const getGroups = jsonRaw => pathOr(null, ['content', 'groups'], jsonRaw);
```

Comment:
No big deal but `path` could be used here instead because `path` returns `undefined` by default if the object path does not exist

```js
import path from 'ramda/src/path';
```

```js
const getGroups = jsonRaw => path(['content', 'groups'], jsonRaw);
```

Reply:
I think this can be simplified further

```js
const getGroups = path(['content', 'groups']);
```

because ramda functions can be optionally curried so that will return a function with jsonRaw as the argument

Reply:
Tried the two options - but they seem to be returning functions and not array of groups as expected. Not sure why exactly